app/models/framework_response.rb

```diff
@@ -42,6 +42,13 @@ def update_with_flags!(value)
     raise ActiveRecord::ReadOnlyRecord, "Can't update framework_responses because person_escort_record is #{person_escort_record.status}"
   end
 
+  def value=(raw_value)
+    unless raw_value.blank? || value_type_valid?(raw_value)
+      errors.add(:value, 'is incorrect type')
+      raise ActiveModel::ValidationError, self
```

Comment:
Discussed offline - liking the simplicity this provides further down the STI tree, but I find it a little strange to be throwing a `ValidationError` without calling `save` or `valid?`. If this was changed to a different exception rather than just adding to the model errors array this would perhaps feel more explicit, and the controller code could catch the exception and convert to a suitable error message for the response.

Ruby isn't typed, and this is essentially a type error - so maybe use `ArgumentError` as the closest approximation to that, or add a custom exception class - and pass in whatever arguments are useful to help build the eventual error to be served to the client.